src/common/components/PlayForms.jsx

```diff
@@ -1,91 +1,123 @@
-                                                    updatedval.push({
-                                                      [field.fieldName || 'name']: v,
-                                                      [field.fieldValue || 'value']: ""
-                                                    })
-                                                  }
-                                                })
-                                              }
-                                              onDataChanged(field.datafield, updatedval)}
-                                            }
-                                            renderInput={(params) => (
-                                              <TextField
-                                                {...params}
-                                                size="small"
-                                                placeholder={field.placeholder}
-                                              />
-                                            )}
-                                />
-                default: return <></>
-    }
-  }
+  const checkDisabledInputs = (fieldName) => isEditPlay && disabledFields.includes(fieldName)
+
+  const handleChange = (key, value) => {
+    setFormData((pre) => ({...pre, [key]: value }));
+  };
+
+  useEffect(() => {
+    setFormData({...formDataObj})
+  },[formDataObj])
  
+  const renderField = (field) => {
+    switch (field.type) {
+      case "input":
+        return (
+          <TextField
+            id={field.id}
+            label={field.plaeholder}
+            value={formData[field.datafield]}
+            size='small'
+            className='w-full'
+            disabled={checkDisabledInputs(field.datafield)}
+            {...field}
+            onChange={(e) => handleChange(field.datafield, e.target.value)}
+          />
+        );
+      case "select":
+        return (
+          <Autocomplete
+            id={field.datafield}
+            size='small'
+            options={field.options || []}
+            getOptionLabel={(option) => option.name || option[field.fieldName] || option}
+            filterSelectedOptions
+            multiple={field.multiple}
+            freeSolo={field.freeSolo}
+            disabled={checkDisabledInputs(field.datafield)}
+            value={formData[field.datafield] ?? ''}
+            onChange={(e, newValue) => {
+              let updatedval = newValue;
+              if (field.multiple) {
+                updatedval = [];
+                newValue.forEach((v) => {
+                  if (_.isObject(v)) {
+                    updatedval.push(v);
+                  } else {
+                    updatedval.push({
+                      [field.fieldName || "name"]: v,
+                      [field.fieldValue || "value"]: "",
+                    });
+                  }
+                });
+              }
+              handleChange(field.datafield, updatedval);
+            }}
```

Comment:
Can we move this logic to a separate function and use that function here?

Reply:
it was there from before so i didnt touch it. im scared if something broke by moving this function out. although i think it has many state setters and function invokation so i gues it will make more complex in order to seperate to a different file.